docs/integrations/argocd-integration.md

```diff
@@ -0,0 +1,411 @@
+      command: ["/bin/sh", "-c"]
+      args:
+        - >
+          cp /etc/github-secret/ssh-private-key /home/terrascan/.ssh/id_ed25519_github &&
+          cp /etc/ssh-config-secret/ssh-config /home/terrascan/.ssh/config &&
+          cp /etc/ssh-known-hosts-secret/ssh-known-hosts /home/terrascan/.ssh/known_hosts &&
+          chmod -R 400 /home/terrascan/.ssh/* &&
+          terrascan server --cert-path /data/server.crt --key-path /data/server.key -p 443 -l debug -c /data/config.toml
+    volumes:
+      - name: certspath
+        hostPath:
+          path: <YOUR CERTIFICATE FOLDER PATH>
+      #add all required ssh keys need to clone your repos
+      - name: ssh-keys-github
+        secret:
+          secretName: github-secret
+      #add a secret for git config file   
+      - name: ssh-config
+        secret:
+          secretName: ssh-config-secret
+      #add a configmap for the ssh known_hosts file
+      - name: ssh-known-hosts
+        configMap:
+          name: known-hosts-config
+```            
+
+For non-public repositories, the private key, known hosts and ssh config needs to be added as a kubernetes secret, configmap and secret respectively.
+  
+```
+kubectl create secret generic github-secret \
+  --from-file=ssh-privatekey= < path to your private key > \
+  --from-file=ssh-publickey=< path to your public key >
+``` 
+
+Config-map: 
+
+``` 
+kubectl create configmap ssh-known-hosts --from-file=< path to you known hosts file >
+``` 
+
+ssh config secret
+
+``` 
+kubectl create secret generic ssh-config-secret \
+  --from-file=< path to your ssh config file >
+``` 
+
+Example ssh config file
+
+``` 
+Host github.com
+  HostName github.com
+  IdentityFile ~/.ssh/id_ed25519_github
+``` 
+
+After making changes to the webhook deployment file, apply this yaml in your cluster.
+
+You can also run terrascan admission controller server outside cluster, for more information on configuring terrascan as an admission controller webhook, follow https://docs.accurics.com/projects/accurics-terrascan/en/latest/integrations/admission-controller-webhooks-usage 
+
+
+#### Step 2: Create a Dockerfile for the container which has the terrascan script to run the remote scan against the terrascan’s admission controller webhook.
+
+```DockerFile
+# Dockerfile with a script to use terrascan's validating webhook
+# configured in the kubernetes cluster, to scan a repo for violations
+FROM alpine:3.12.0
+
+#curl to send request to terrascan validating webhook
+RUN apk add --no-cache curl 
+
+WORKDIR /home/terrascan
+
+RUN mkdir bin
+
+COPY scripts/argocd-terrascan-remote-scan.sh  bin/terrascan-remote-scan.sh
+
+# create non root user
+RUN addgroup --gid 101 terrascan && \
+  adduser -S --uid 101 --ingroup terrascan terrascan
+
+RUN chown -R terrascan:terrascan bin && \
+  chmod u+x bin/terrascan-remote-scan.sh
+
+USER terrascan
+
+CMD ["sh"]
+```
+
+terrascan-remote-scan script
+
+```sh
+#!/bin/sh
+TERRASCAN_SERVER="https://${SERVICE_NAME}"
```

Comment:
-we should make the script more robust.
-we can handle cases where some required env variables are not provided. (we should print a message and exit non zero)
-also, we should make the vars configurable, so that (if required) we would be able to extend the scan to other iac's and types as well

Reply:
added check for required env variables.